### PR TITLE
sstables/metadata_collector: drop unused function

### PR DESCRIPTION
**Pull request by tchaikov**
column_stats::update_local_deletion_time() is not used anywhere, what is being used is `column_stats::update_local_deletion_time_and_tombstone_histogram()`. so let's drop the former.

### CODE REVIEW

**Comment by bhalevy** [comment]
It might be more elegant to have it return int32_t
and drop `update_local_deletion_time_and_tombstone_histogram(int32_t value)` instead,
so that `update_local_deletion_time_and_tombstone_histogram(gc_clock::time_point value)` can use it as a building block to update local_deletion_time and use the returned, adjusted value to update `tombstone_histogram`

**Comment by tchaikov** [comment]
i went further by removing both of them.

File: sstables/metadata_collector.hh [hunk 1]
```diff
@@ -94,11 +85,6 @@ struct column_stats {
     void update_ttl(gc_clock::duration value) {
         ttl_tracker.update(gc_clock::as_int32(value));
     }
-    void update(const deletion_time& dt) {
-        assert(!dt.live());
-        update_timestamp(dt.marked_for_delete_at);
```
Comment (bhalevy) [inline sstables/metadata_collector.hh:99]: that's weird.
what function tracks this timestamp then?

Comment (bhalevy) [reply sstables/metadata_collector.hh:99]: This should be called on the memtable write path to update the encoding stats in the `encoding_stats_collector`.
The minimum values are used when encoding delta-values (like in `do_write_delta_deletion_time`) when flushing the memtable.

In compaction we combine all sstable encoding stats into `_stats_collector` and use that to encode the output sstable (that said, we don't take deletions into account so it's possible the minimum values can theoretically be be fine-tuned, but we don't do that)

Comment (bhalevy) [reply sstables/metadata_collector.hh:99]: Looks like `memtable::memtable_encoding_stats_collector::update(tombstone tomb)` called from `memtable::memtable_encoding_stats_collector::update(const ::schema& s, const mutation_partition& mp)` for `mp.partition_tombstone()` is doing that.

Comment (tchaikov) [reply sstables/metadata_collector.hh:99]: @bhalevy hi Benny, as `column_stats` is solely used by `sstables::sstable_writer::writer_impl` . so i am trying to focus on the code path which involves this class. upon finishing processing a partition, we collect `column_stats` which tracks the columns in the last partition into `writer::_collector`. see `writer::consume_end_of_partition()`.

i was trying to explain the reason why `deletion_time.marked_for_delete_at` was already tracked in my previous comment. but seems i failed to do so. please allow me to rephrase it.

`deletion_time` is a part of the on-disk format. if we wanted to track `deletion_time.marked_for_delete_at`, intuitively, we would track it using its internal representation. in the write path, as you put, the corresponding internal representation of `deletion_time` is `tombstone`. and the corresponding field is `tombstone::timestamp`. so, in the write path, when we consume a tombstone, the `tombstone::timestamp` is tracked by `column_stats.timestamp_tracker`.  see `writer::consume(tombstone t)`. guess that's why we never update `column_stats` with `deletion_time` . the stats will be included as part of the stats component when sealing the sstable.

but yeah, we also track the `encoding_stats` when flushing memtable to sstable, or when compacting sstables. but this is orthogonal to the use case of `column_stats` which i am trying to cleanup. as you explained above, the `tombstone::timestamp` is taken care of by `memtable::memtable_encoding_stats_collector::update(tombstone)`. and the min timestamp recorded in the collected stats is used the baseline when persisting the timestamps as deltas.

anyway, i think these two stats serve two different purposes. am i right?

could you please take another look at your convenience?

**Review by bhalevy — APPROVED** [review]

File: sstables/metadata_collector.hh (hunk 1 above)
Comment (bhalevy) [reply sstables/metadata_collector.hh:99]: Yes, they are used for different purposes.
Can you please verify that we correctly clamp min/max local deletion time in the encondig stats? IIRC they are stored as int32_t too in the sstable, but the memtable stats tracker is different thank the sstable's.

Comment (tchaikov) [reply sstables/metadata_collector.hh:99]: sure. will get back to you in this thread.

Comment (tchaikov) [reply sstables/metadata_collector.hh:99]: @bhalevy , only `encoding_stats.min_local_deletion_time` is persisted. this field is used to update `encoding_stats_collector.min_local_deletion_time`. see [encoding_stats_collector::update()](https://github.com/scylladb/scylladb/blob/4eb4ac4634071341dedee6a54b5bb464567b832e/encoding_stats.hh#L68) . and `encoding_stats_collector.min_local_deletion_time` is in turn stored in `sstable_schema.header.min_local_deletion_time_base`, see [`make_sstable_schema()`](https://github.com/scylladb/scylladb/blob/4eb4ac4634071341dedee6a54b5bb464567b832e/sstables/mx/writer.cc#L389). please note, it is casted to a `vint<uint64_t>`. so we do *not* support the time before UNIX epoch. this reminds me the discussion on the [design doc of Capping local_deletion_time](https://docs.google.com/document/d/1Lo0nqQPWNl1Dc-K9WK0hGkEGxZSJkv1ZU38JFlxGN7o/edit?usp=sharing) .

and the header is then serialized with `statistics.contents[metadata_type::Serialization]` . see [writer::consume_end_of_stream()](https://github.com/scylladb/scylladb/blob/4eb4ac4634071341dedee6a54b5bb464567b832e/sstables/mx/writer.cc#L1455).

so, in short, `min_local_deletion_time` is stored as `vint<uint64_t>`.

Comment (bhalevy) [reply sstables/metadata_collector.hh:99]: OK.  BTW, `get_min_local_deletion_time()` converts it into a signed `int64_t` around the Cassandra epoch time `deletion_time_epoch` (22/09/2015, 00:00:00)

**Comment by bhalevy** [comment]
@scylladb/scylla-maint please review/merge
